debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/connection/PostgresReplicationConnection.java

```diff
@@ -445,50 +485,112 @@ else if (e.getMessage().matches("(?s)ERROR: requested WAL segment .* has already
             private ExecutorService keepAliveExecutor = null;
             private AtomicBoolean keepAliveRunning;
             private final Metronome metronome = Metronome.sleeper(statusUpdateInterval, Clock.SYSTEM);
+            private ReplicationMessageProcessor processor;
+            private long nextSeq = 0;
 
-            // make sure this is volatile since multiple threads may be interested in this value
-            private volatile LogSequenceNumber lastReceivedLsn;
+            @Override
+            public void init(ReplicationMessageProcessor processor) {
+                this.processor = processor;
+                LOGGER.info("Starting processing threads = {}", receiveQueueParallelism);
+                processorThreadPool = new ThreadPoolExecutor(
+                        receiveQueueParallelism, receiveQueueParallelism, 0, TimeUnit.MILLISECONDS,
+                        new LinkedBlockingQueue());
+                for (int i = 0; i < receiveQueueParallelism; i++) {
+                    processorThreadPool.execute(() -> {
+                        LOGGER.info("Starting thread to process from receive queue");
+                        while (true) {
+                            List<RawReplicationMessage> records;
+                            try {
+                                records = receiveQueue.poll();
+                            }
+                            catch (InterruptedException ex) {
+                                LOGGER.warn("Processor thread interrupted, stopping");
+                                break;
+                            }
+                            if (records.isEmpty()) {
+                                continue;
+                            }
+                            BufferingChangeEventQueue.startBuffering();
+                            for (RawReplicationMessage msg : records) {
+                                try {
+                                    deserializeMessages(msg);
+                                }
+                                catch (InterruptedException ex) {
+                                    LOGGER.warn("Processor thread interrupted, stopping");
+                                    break;
+                                }
+                                catch (SQLException ex) {
+                                    // We cannot simply ignore this failure and move to the next message
+                                    // as that may violate ordering and exactly-once semantics.
+                                    LOGGER.warn("Processor thread exception:", ex);
+                                    throw new RuntimeException("Exception in reading from stream: " + ex.getMessage());
```

Comment:
Does this restart the task? All other threads will be waiting for the nextSeq for them to proceed.

Reply:
Not sure what the right thing to do is.  Added a TODO.